preliz/internal/parser.py

```diff
@@ -13,8 +15,18 @@ def inspect_source(fmodel):
     source = inspect.getsource(fmodel)
     signature = inspect.signature(fmodel)
     source = re.sub(r"#.*$|^#.*$", "", source, flags=re.MULTILINE)
-
-    return source, signature
+    default_params = {
+        name: (param.default if param.default is not inspect.Parameter.empty else np.nan)
+        for name, param in signature.parameters.items()
+    }
+    model = fmodel(**default_params)
+    if isinstance(model, pm.Model):
+        engine = "pymc"
+    elif isinstance(model, bmb.Model):
```

Comment:
This is the right thing to do in general, but this will require the user to have pymc and bambi to be installed.